### PR TITLE
Issue/187 theme use svg logos

### LINKED ISSUE #187: Use an SVG for default theme image/lockup for az_barrio
## Conditions of satisfaction
- [ ] Default header logo is not a PNG, and is now an SVG.
- [ ] Default footer logo is not a PNG, and is now an SVG.

Advantages:
Higher resolution with vector graphics
More easily scalable
Smaller file size
Drupal 8 expects SVG by default.

### PR DESCRIPTION
**Pull request by trackleft**
## Description
[Added a patch to allow svgs during form validation.](https://www.drupal.org/project/drupal/issues/2259567)
@kevcooper gave the ok to allow adding SVG on theme settings form, as long as we lock down permissions with regard to who can use that form, and should treat it the same way we do uploading Javascript.
Additionally, found that the existing footer logo was not functional at all, so I fixed that in order to upload an SVG there.

## Related Issue
Closes #187 

## How to test.
1. go to /admin/appearance/settings/az_barrio
2. Find Logo image section and determine default image.
3. Verify the site uses the default image by default.
4. Upload an SVG
5. Verify the site uses the SVG
6. toggle the Inline logo SVG option and verify the new option is reflected in the HTML.
7. Follow same steps for Footer Logo Image

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Checklist:
- [ ] I've added this Pull Request to the AZ Quickstart project in the right column.
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by trackleft** [comment]
After spending time checking out Drupal Core, this theme needs some serious refactoring in order to more easily add functionality.

File: themes/custom/az_barrio/theme-settings.php [hunk 1]
```diff
@@ -259,10 +282,20 @@ function az_barrio_form_system_theme_settings_alter(&$form, FormStateInterface $
   // Footer logo.
   $form['footer_logo'] = [
     '#type' => 'details',
-    '#title' => t('Logo Footer Image'),
+    '#title' => t('Footer Logo Image'),
     '#collapsible' => TRUE,
     '#collapsed' => TRUE,
   ];
+  $form['footer_logo']['az_barrio_footer_logo_svg_inline'] = [
+    '#type' => 'select',
```
Comment (trackleft) [inline themes/custom/az_barrio/theme-settings.php:290]:
```suggestion
    '#type' => 'checkbox',
```

File: themes/custom/az_barrio/theme-settings.php [hunk 2]
```diff
@@ -259,10 +282,20 @@ function az_barrio_form_system_theme_settings_alter(&$form, FormStateInterface $
...
+    '#title' => t('Inline logo SVG'),
+    '#options' => [
+      1 => t('On'),
+      0 => t('Off'),
+    ],
```
Comment (trackleft) [inline themes/custom/az_barrio/theme-settings.php:295]:
```suggestion
```

File: themes/custom/az_barrio/theme-settings.php [hunk 3]
```diff
@@ -236,6 +248,17 @@ function az_barrio_form_system_theme_settings_alter(&$form, FormStateInterface $
...
     '#default_value' => theme_get_setting('az_barrio_navbar_offcanvas'),
   ];
+  // Logos.
+  $form['logo']['az_barrio_logo_svg_inline'] = [
+    '#type' => 'select',
+    '#title' => t('Inline logo SVG'),
+    '#options' => [
+      1 => t('On'),
+      0 => t('Off'),
+    ],
```
Comment (trackleft) [inline themes/custom/az_barrio/theme-settings.php:258]:
```suggestion
    '#type' => 'checkbox',
    '#title' => t('Inline logo SVG'),
```

File: themes/custom/az_barrio/config/schema/az_barrio.schema.yml [hunk 4]
```diff
@@ -235,9 +235,12 @@ az_barrio.settings:
...
       type: boolean
       label: 'Enable AZ Bootstrap sticky footer template'
-    use_material_design_sharp_icons:
+    az_barrio_logo_svg_inline:
       type: boolean
-      label: 'Enable Material Design Sharp Icons'
+      label: 'Inline logo SVG'
+    az_barrio_footer_logo_svg_inline:
+      type: boolean
+      label: 'Inline footer logo SVG '
```
Comment (trackleft) [inline themes/custom/az_barrio/config/schema/az_barrio.schema.yml:243]:
```suggestion
      label: 'Inline footer logo SVG'
```

File: themes/custom/az_barrio/theme-settings.php [hunk 5]
```diff
@@ -259,10 +282,20 @@ function az_barrio_form_system_theme_settings_alter(&$form, FormStateInterface $
...
   $form['footer_logo'] = [
     '#type' => 'details',
-    '#title' => t('Logo Footer Image'),
+    '#title' => t('Footer Logo Image'),
     '#collapsible' => TRUE,
     '#collapsed' => TRUE,
   ];
+  $form['footer_logo']['az_barrio_footer_logo_svg_inline'] = [
+    '#type' => 'select',
+    '#title' => t('Inline logo SVG'),
```
Comment (trackleft) [inline themes/custom/az_barrio/theme-settings.php:291]:
```suggestion
    '#title' => t('Inline footer logo SVG'),
```

**Review by smayden — APPROVED** [review]
For reals this time:
-Verified that png default image now appears (other reviewers may need to rebuild to see the defaults)
-Was able to add SVG to both logo and footer logo
-Toggled on/off for inline SVG option and saw the expected changes when inspecting HTML
-Verified that new checkbox for inline SVG option functions as expected.

**Review by camikazegreen — APPROVED** [review]

**Pull request merged**